### PR TITLE
chore: update GNU Arm toolchain to 14.2.rel1

### PR DESCRIPTION
**Pull request by pfeerick**
Summary of changes:
- update to arm-gnu-toolchain-14.2.rel1
- bring EdgeTX out of the C++ stone age (C++17)  (ok, maybe not :laughing:)

Bringing this up for increased visibility and testing... ~~depends on EdgeTX/build-edgetx#23~~

### CODE REVIEW

File: radio/src/CMakeLists.txt [hunk 1]
```diff
@@ -120,6 +120,9 @@ endif()
 
 if(ARCH STREQUAL ARM)
   include(targets/common/arm/CMakeLists.txt)
+  if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "13.2.1")
+    message(FATAL_ERROR "Only arm gnu toolchain 13.2.rel1 is currently supported")
```
Comment (pfeerick) [inline radio/src/CMakeLists.txt:124]: I personally am probably of the same opinion as Raphael... let people be idiots and not use this toolchain... i.e. make it warning, but not fatal

**Comment by pfeerick** [comment]
Tests will fail if FATAL_ERROR

**Comment by philmoz** [comment]
Why 13.2 and not 13.3 or 14.2?

**Comment by gagarinlg** [comment]
Because that was the release that was current, when I started this a while ago. Since 14.2 is current now, I am happy to use that

**Comment by philmoz** [comment]
Version check should probably be updated to 14.2.1 as well.

**Comment by elecpower** [comment]
Compiling the libsims now report
> CMake Warning at radio/src/CMakeLists.txt:119 (message):
  Only Arm GNU Toolchain version 14.2.rel1 is supported

**Comment by pfeerick** [comment]
Yeah, it is possible it is reporting different versions for different OSes... i.e. instead of 14.2.1 ... will poke around a bit more to see what is going on, otherwise Malte's version warning goes bye bye... it got changed from a fatal error to a warning 6 months ago as it is 😆 

msys
```
CMake Warning at radio/src/CMakeLists.txt:119 (message):
  Only Arm GNU Toolchain version 14.2.rel1 is supported
TARGET simu/libsimulator: cpp compiler /usr/bin/c++ v9
```

linux
```
CMake Warning at radio/src/CMakeLists.txt:119 (message):
  Only Arm GNU Toolchain version 14.2.rel1 is supported
TARGET simu/libsimulator: cpp compiler D:/a/_temp/msys64/mingw64/bin/c++.exe v14.2.0
```

**Pull request merged**